design/src/rfcs/rfc0041_file_per_change_changelog.md

```diff
@@ -0,0 +1,125 @@
+RFC: File-per-change changelog
+==============================
+
+> Status: RFC
+>
+> Applies to: client and server
+
+For a summarized list of proposed changes, see the [Changes Checklist] section.
+
+Historically, the smithy-rs and AWS SDK for Rust's changelogs and release notes have been
+generated from the `changelogger` tool in `tools/ci-build/changelogger`. This is a tool built
+specifically for development and release of smithy-rs, and it requires developers to add
+changelog entries to a root `CHANGELOG.next.toml` file. Upon release, the `[[smithy-rs]]` entries
+in this file go into the smithy-rs release notes, and the `[[aws-sdk-rust]]` entries are associated
+with a smithy-rs release commit hash, and added to the `aws/SDK_CHANGELOG.next.json` for
+incorporation into the AWS SDK's changelog when it releases.
+
+This system has gotten us far, but it has always made merging PRs into main more difficult
+since the central `CHANGELOG.next.toml` file is almost always a merge conflict for two PRs
+with changelog entries.
+
+This RFC proposes a new approach to change logging that will remedy the merge conflict issue,
+and explains how this can be done without disrupting the current release process.
+
+The proposed developer experience
+---------------------------------
+
+There will be a `changelog/` directory in the smithy-rs root where
+developers can add changelog entry Markdown files. Any file name can be picked
+for these entries. Suggestions are the development branch name for the
+change, or the PR number.
+
+The changelog entry format will change to make it easier to duplicate entries
+across both smithy-rs and aws-sdk-rust, a common use-case.
+
+This new format will make use of Markdown front matter in the YAML format.
+This change in format has a couple benefits:
+- It's easier to write change entries in Markdown than in a TOML string.
+- There's no way to escape special characters (such as quotes) in a TOML string,
+  so the text that can be a part of the message will be expanded.
+
+While it would be preferable to use TOML for the front matter (and there are libraries
+that support that), it will use YAML so that GitHub's Markdown renderer will recognize it.
+
+A changelog entry file will look as follows:
+
+```markdown
+---
+# Adding `aws-sdk-rust` here duplicates this entry into the SDK changelog.
+applies_to: ["client", "server", "aws-sdk-rust"]
+authors: ["author1", "author2"]
+references: ["smithy-rs#1234", "aws-sdk-rust#1234"]
+# The previous `meta` section is broken up into its constituents:
+breaking: false
+# This replaces "tada":
+new_feature: false
+bug_fix: false
```

Comment:
you may want some sort of ordering field if you want any sort of consistent ordering during change-log rendering (but maybe it's fine to just go alphabetically?)

Reply:
I think we may need to experiment with this to get a good balance between rendered result and dev experience. Two ideas that come to mind to try:
1. Add an ordering field that takes a RFC-3339 date. Have a CLI that creates new changelog entries and automatically fills this in.
2. Have some kind of prominence enum and field to categorize items as more prominent than others.